funsor/terms.py

```diff
@@ -431,28 +430,74 @@ def __max__(self, other):
         return Binary(ops.max, self, to_funsor(other))
 
     def __getitem__(self, other):
-        return Binary(ops.getitem, self, to_funsor(other))
+        if type(other) is not tuple:
```

Comment:
Hmm sure, we can support all PyTorch operations, but `.unsqueeze()` would be implemented separately from `__getitem__()`:
```py
class Tensor(Funsor):
    def unsqueeze(dim):
        if dim >= 0:
            dim -= len(self.output.shape)
        return Tensor(self.data.unsqueeze(dim), self.inputs, self.dtype)
```
Note that unsqueeze operates only on the `.output` shape, not in inputs. There is no need for for `.unsqueeze()` on funsor inputs because inputs match by name rather than position.

Reply:
That makes sense! I meant using None as an index for the `unsqueeze` operation which torch/numpy support, but I'm not sure if it is feasible to add it in `GetItemOp` or even worth it. I think having a separate method for that seems like a simpler solution.

```python
>>> t = torch.randn(2, 3)
>>> t[..., None, None].shape
torch.Size([2, 3, 1, 1])
```